crates/jit/src/instantiate.rs

```diff
@@ -38,9 +38,17 @@ pub struct CompiledModuleInfo {
     /// Sorted list, by function index, of names we have for this module.
     func_names: Vec<FunctionName>,
 
-    /// The trampolines compiled into the text section and their start/length
-    /// relative to the start of the text section.
-    pub trampolines: Vec<(SignatureIndex, FunctionLoc)>,
+    /// Metadata about array-to-Wasm trampolines. Used when exposing a Wasm
+    /// callee to an array caller.
+    array_to_wasm_trampolines: Vec<(DefinedFuncIndex, FunctionLoc)>,
+
+    /// Metadata about native-to-Wasm trampolines. Used when exposing a Wasm
+    /// callee to a native caller.
+    native_to_wasm_trampolines: Vec<(DefinedFuncIndex, FunctionLoc)>,
```

Comment:
To cut down on the encoding space necessary for this, could these two arrays get fused? And to go a bit further could this perhaps be stored in the `funcs` array directly?